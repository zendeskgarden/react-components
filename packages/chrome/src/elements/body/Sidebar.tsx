/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React, { HTMLAttributes } from 'react';
import { StyledSidebar } from '../../styled';

/**
 * @extends HTMLAttributes<HTMLElement>
 */
export const Sidebar = React.forwardRef<HTMLElement, HTMLAttributes<HTMLElement>>((props, ref) => (
  <StyledSidebar ref={ref} {...props} />
));

Sidebar.displayName = 'Sidebar';
