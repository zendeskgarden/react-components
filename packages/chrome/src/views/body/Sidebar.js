import styled from 'styled-components';
import { retrieveTheme } from '@zendesk/garden-react-theming';
import ChromeStyles from '@zendesk/garden-css-chrome';

import { version } from '../../../package.json';
const COMPONENT_ID = 'chrome.sidebar';

/**
 * Accepts all `<aside>` props
 */
const Sidebar = styled.aside.attrs({
  'data-garden-id': COMPONENT_ID,
  'data-garden-version': version,
  className: ChromeStyles['c-chrome__body__content__sidebar']
})`
  ${props => retrieveTheme(COMPONENT_ID, props)};
`;

/** @component */
export default Sidebar;
