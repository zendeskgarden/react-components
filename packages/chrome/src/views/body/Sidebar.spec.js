import React from 'react';
import { shallow } from 'enzyme';

import Sidebar from './Sidebar';

describe('Sidebar', () => {
  it('renders default styling', () => {
    const wrapper = shallow(<Sidebar />);

    expect(wrapper).toMatchSnapshot();
  });
});
