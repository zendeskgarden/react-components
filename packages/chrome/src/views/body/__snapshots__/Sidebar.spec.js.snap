// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders default styling 1`] = `
<aside
  className="c-chrome__body__content__sidebar "
  data-garden-id="chrome.sidebar"
  data-garden-version="version"
/>
`;
